### ISSUE TITLE
twisted/test/test_newcred.py should use setUp and tearDown instead of setUpClass and tearDownClass

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2745]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2745|
|Type|task|
|Created|2007-07-20 14:41:55Z|

PluggableAuthenticationModulesTest hooks the pamauth module's callIntoPAM function in setUpClass and restores it in tearDownClass.  There's no reason not to do this in setUp and tearDown, instead.

<details><summary>Searchable metadata</summary>

```
trac-id__2745 2745
type__task task
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1184942515000000 1184942515000000
changetime__1185626651000000 1185626651000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Done in setup-teardown-newcred-2745

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Fine please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [20857]) Merge setup-teardown-newcred-2745

Author: exarkun
Reviewer: therve
Fixes [#2745](https://github.com/twisted/twisted/issues/2745)

Replace setUpClass and tearDownClass, which mocked the callIntoPAM
global of the pamauth module, with setUp and tearDown methods which
accomplish the same task.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
